payjoin-cli/src/app.rs

```diff
@@ -291,22 +291,23 @@ impl App {
         log::trace!("check3");
 
         // Receive Check 4: have we seen this input before? More of a check for non-interactive i.e. payment processor receivers.
-        let payjoin = proposal.check_no_inputs_seen_before(|input| {
-            Ok(!self.insert_input_seen_before(*input).map_err(|e| Error::Server(e.into()))?)
-        })?;
+        let mut provisional_payjoin = proposal.check_no_inputs_seen_before(
+            |input| {
+                Ok(!self.insert_input_seen_before(*input).map_err(|e| Error::Server(e.into()))?)
+            },
+            |output_script| {
+                if let Ok(address) = bitcoin::Address::from_script(output_script, network) {
+                    self.bitcoind
+                        .get_address_info(&address)
+                        .map(|info| info.is_mine.unwrap_or(false))
+                        .map_err(|e| Error::Server(e.into()))
+                } else {
+                    Ok(false)
+                }
+            },
+        )?;
```

Comment:
identify_receiver_outputs does not seem directly related to the check, so I think we lose some semantics by merging them into one step. Either way it's unrelated to the ProvisionalProposal struct and would be better suited to its own PR

Reply:
yea I see what u mean
maybe we can change the parent function name

Reply:
I don't think it should be renamed rather not combined. It's unrelated to this PR.